func.go

```diff
@@ -382,13 +386,17 @@ func (f *baseFuncObject) hasInstance(v Value) bool {
 				}
 			}
 		} else {
-			f.val.runtime.typeErrorResult(true, "prototype is not an object")
+			val.runtime.typeErrorResult(true, "prototype is not an object")
```

Comment:
So the net result here will still panic [here](https://github.com/mongodb-forks/goja/blob/5b0ecee1e091bebd15e46c9722e634b9a80c6fc9/runtime.go#L478). I was basing not pulling over this change since it looked like most other areas in our code preferred the `typeErrorResult` method instead of the explicit panic. I'll admit I do like the upstream more 😄

Reply:
The template-backed objects commit changes this to a panic. The logic is the same, but I think we should match upstream to make future cherry-picks have less conflicts.

Reply:
Got it 👍 I'll update these instances to follow upstream